includes/data-stores/class-wc-coupon-data-store-cpt.php

```diff
@@ -210,6 +210,9 @@ private function update_post_meta( &$coupon ) {
 				case 'email_restrictions' :
 					$updated = update_post_meta( $coupon->get_id(), $meta_key, array_filter( array_map( 'sanitize_email', $value ) ) );
 					break;
+				case 'date_expires' :
+					$updated = update_post_meta( $coupon->get_id(), $meta_key, ( $value ? $value->date( 'Y-m-d' ) : '' ) );
```

Comment:
@mikejolley I've gone through and updated things to use local time where appropriate. I believe this should store in local time, as people are picking the date in local time.

This is how I see it working. Please let me know if I'm mistaken.
1. Select/Load date as 'Y-m-d' in local time
2. 'Y-m-d' gets converted to WC_DateTime in the appropriate timezone automatically when `set_date_expires` happens.
3. Convert the object back into 'Y-m-d' in local time on save